asab/library/providers/zookeeper.py

```diff
@@ -340,15 +348,53 @@ def recursive_traversal(path, digest):
 
 
 	async def _on_library_changed(self, event_name=None):
+		"""
+		Check watched paths and publish a pubsub message for every one that has changed.
+		"""
 
-		for path, digest in self.Subscriptions.items():
+		for path, digests in self.Subscriptions.items():
+
+			async def do_check_path(tenant=None):
+				# If tenant is None, we are checking global layer
+				if tenant is not None:
+					path = "/.tenants/{}{}".format(tenant, path)
+				try:
+					newdigest = await self._get_directory_hash(path)
+				except kazoo.exceptions.NoNodeError:
+					# This node is either deleted or has never existed.
+					newdigest = None
+
+				if newdigest != digests.get(tenant):
+					self.Subscriptions[path][tenant] = newdigest
+					ctx = Tenant.set(tenant)
```

Comment:
This will not work for sync->async PubSub

Reply:
context will be replaced with `target` argument

tenant targets:
```python
self.App.PubSub.publish("Library.change!", self, path, target=("tenant", tenant))
```

global targets:
```python
self.App.PubSub.publish("Library.change!", self, path)  # For backward compatibility
```